### DISCUSSION TITLE
Anyone do a step by step on a Docker install on MacOSX?

### DISCUSSION THREAD

**Discussion by markss63** [philosowaffle/peloton-to-garmin#205] [Q&A]
I'm a total docker newb.  I had p2g_v1 and used to use it, but now trying to make this new version work.  Has anyone done a more detailed step by step?

I installed Docker desktop, got images, and am trying to run things, but I can't even find where things are on my computer!

I managed to generate and find a log file, but it looks like my config file is in the wrong place?  it pretty much errors out immediately

Anyone can help?

# more log20220124.txt
2022-01-24 16:00:00.468 +00:00 [ERR] Failed to get syncServiceStatus from db
System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at JsonFlatFileDataStore.DataStore.GetItem[T](String key)
   at Common.Database.SyncStatusDb.GetSyncStatusAsync() in /build/src/Common/Database/SyncStatusDb.cs:line 38
2022-01-24 16:00:00.694 +00:00 [INF] Application started. Press Ctrl+C to shut down.
2022-01-24 16:00:00.696 +00:00 [INF] Hosting environment: Production
2022-01-24 16:00:00.697 +00:00 [INF] Content root path: /app
2022-01-24 16:00:01.544 +00:00 [ERR] Http Call Failed. "Unauthorized" {"status":401,"error_code":3010,"subcode":null,"message":"Login 
failed","details":null}
2022-01-24 16:00:01.556 +00:00 [FTL] Failed to authenticate with Peloton.
Flurl.Http.FlurlHttpException: Call failed with status code 401 (Unauthorized): POST https://api.onepeloton.com/auth/login
   at Flurl.Http.FlurlRequest.HandleExceptionAsync(FlurlCall call, Exception ex, CancellationToken token)
   at Flurl.Http.FlurlRequest.SendAsync(HttpMethod verb, HttpContent content, CancellationToken cancellationToken,

**Comment by philosowaffle** [comment]
I don't have a mac, but I might still be able to help a bit.

First thing to note, you can still run version 2 from the command if you would like. But migrating to Docker is a great way to go if you're willing to do it.

Jump down to the "File Sharing" section on [this link](https://docs.docker.com/desktop/mac/). We need to share some files with the P2G container.

Let's assume you have the following folder on your mac: `/Users/me/docker/p2g/` And in this folder you also have your config file `configuration.local.json`.

Following the instructions on that link, configure Docker desktop to share the following paths, the paths on the left are your Mac and the path on the right is the path inside the container we want to map to.

```
/Users/me/docker/p2g => /app
```

In laymans terms, this creates a sort of shared folder that both your Mac and the docker container can access. But it only works if the file path specified in the container (the right hand side path) actually exists and is being used by the application.

Now when the container runs it should be able to read the config file from the shared directory and it should be able to write output back into the directory. In your p2g config file, leave the [working and output paths](https://github.com/philosowaffle/peloton-to-garmin/blob/master/configuration.example.json#L4) as they are in this example and they should also show up in your local Mac p2g folder.
